File: src/routes/authRoutes.js
const express = require('express');
const router = express.Router();
const { signup, signin, getProfile } = require('../controllers/authController');
const { protect } = require('../middleware/auth');
const {
  validateSignup,
  validateSignin,
  handleValidationErrors,
} = require('../middleware/validate');

router.post('/signup', validateSignup, handleValidationErrors, signup);
router.post('/signin', validateSignin, handleValidationErrors, signin);
router.get('/profile', protect, getProfile);

module.exports = router;

File: auth-service-go/routes/auth_routes.go
package routes

import (
	"auth-service-go/controllers"
	"auth-service-go/middleware"

	"github.com/gofiber/fiber/v2"
)

func SetupAuthRoutes(app *fiber.App) {
	api := app.Group("/api/auth")

	api.Post("/signup", middleware.ValidateSignup, controllers.Signup)
	api.Post("/signin", middleware.ValidateSignin, controllers.Signin)
	api.Get("/profile", middleware.Protect, controllers.GetProfile)
}
